### PR TITLE
Convert 'user-ad-targeting.js' to ES6

### PR DESCRIPTION
**Pull request by lps88**
ES6 ✨ 

Only real change here is 
```
for (key in user.adData) {
    userSegments.push(key + user.adData[key]);
}
```
is now:
```
Object.keys(user.adData).forEach(key => {
    userSegments.push(key + user.adData[key]);
});
```
because ESLint.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/user-ad-targeting.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// @flow
+import { local } from 'lib/storage';
+import id from 'common/modules/identity/api';
+
+const userSegmentsKey = 'gu.ads.userSegmentsData';
+
+const getUserSegments = function() {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/user-ad-targeting.js]: Can you specify the return type?

File: static/src/javascripts/projects/commercial/modules/user-ad-targeting.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                return userSegmentsData.segments;
+            }
+            local.remove(userSegmentsKey);
+        }
+    }
+
+    return [];
+};
+
+const requestUserSegmentsFromId = function() {
```
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/user-ad-targeting.js]: ditto

**Comment by janua** [comment]
Odd that you can't use `for (key in user.adData)`

:+1:

**Review by lps88 — COMMENTED** [review]
@janua Yeah it gives an error:
`for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array`

**Comment by janua** [comment]
Ah that makes sense!

**Pull request merged**